agent/proxycfg/api_gateway.go

```diff
@@ -461,3 +462,49 @@ func (h *handlerAPIGateway) watchIngressLeafCert(ctx context.Context, snap *Conf
 
 	return nil
 }
+
+func (h *handlerAPIGateway) generateIngressDNSSANs(snap *ConfigSnapshot) []string {
```

Comment:
So, this is only used for generating additional SANs on the leaf certificate used for a TLS-based listener. Since we provision TLS independently of the leaf cert and only use the cert with its spiffe id for encrypting/verifying mesh traffic, we shouldn't need any of these SANs, so you should be able to drop this entire function and its call above.

Reply:
Cool, I actually wasn't 100% sure where this came in from. I saw you added it and didn't want to nuke something accidentally.